### ISSUE TITLE
Unique decorator that incorporates `deletedAt` field

### ISSUE THREAD

**Issue by goldcaddy77** [goldcaddy77/warthog#223]

**Comment by goldcaddy77** [comment]
Note that soft deletes landed in TypeORM recently.  Just need to upgrade: https://github.com/typeorm/typeorm/pull/5034

**Comment by ghaedi1993** [comment]
i make it work ,manually by adding partial index from inside postgresql as follows 
```

CREATE UNIQUE INDEX index_name on users (name)
WHERE ("deletedAt" IS null)
```

   ```
@Column()
    @Index(["name"],{unique:true,where:"\"createdAt\" IS null"})
    name: string
```

is not resulting the same, makes an index in database but not imposing any unique policy . feature's still not implemented or im getting it wronge ?

**Comment by goldcaddy77** [comment]
You're right - the feature is still not implemented.  Your solution is interesting.  There is no reason that I couldn't create my own Unique decorator that adds this type of index.  I'm going to look into whether TypeORM has come up with an elegant solution for this as it seems like a pretty common problem.
